open-apparel-registry/open-apparel-registry
# Wrap facility search results with react-infinite via react-infininte-any-height

## Overview

Wraps search facility results with `react-infinite` via `react-infininte-any-height` in preparation for loading more than 500 results.

Connects #705

## Demo

![2019-07-30 10 18 17](https://user-images.githubusercontent.com/17363/62150638-67179680-b2b3-11e9-8823-374c470fb92f.gif)


## Notes

React infinite requires that we specify the height of the list item container. We accomplish this by listening for window resize events, tracking the window size in the Redux state, then calculating the proper container size.

`react-infinite-any-height` is a wrapper around `react-infinite` that wraps each list item with a span and uses that span to calculate the height, preventing the need to force all list items to be the same height. There is a lot of variability in the length of facility names and addresses. On narrow screens some list items can be twice as tall as others.

## Testing Instructions

* Run `./scripts/update` to install the new packages
* Verify that the facility search UI works as before

## Checklist
